Source/NSGridView.m

```diff
@@ -0,0 +1,639 @@
+/* Implementation of class NSGridView
+   Copyright (C) 2020 Free Software Foundation, Inc.
+   
+   By: Gregory John Casamento
+   Date: 08-08-2020
+
+   This file is part of the GNUstep Library.
+   
+   This library is free software; you can redistribute it and/or
+   modify it under the terms of the GNU Lesser General Public
+   License as published by the Free Software Foundation; either
+   version 2.1 of the License, or (at your option) any later version.
+   
+   This library is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+   Lesser General Public License for more details.
+   
+   You should have received a copy of the GNU Lesser General Public
+   License along with this library; if not, write to the Free
+   Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+   Boston, MA 02110 USA.
+*/
+
+#import <Foundation/NSArray.h>
+#import "AppKit/NSGridView.h"
+
+#import "GSFastEnumeration.h"
+
+@interface NSGridRow (Private)
+- (void) _setRow: (NSMutableArray *)row;
+- (NSMutableArray *) _row;
+@end
+
+@interface NSGridColumn (Private)
+- (void) _setColumn: (NSMutableArray *)col;
+- (NSMutableArray *) _column;
+@end
+
+@implementation NSGridView
+
+- (instancetype) initWithFrame: (NSRect)frameRect
+{
+  self = [super initWithFrame: frameRect];
+  if (self != nil)
+    {
+      _rows = [[NSMutableArray alloc] initWithCapacity: 10];
+    }
+  return self;
+}
+
+- (instancetype) initWithViews: (NSArray *)rows
+{
+  self = [self initWithFrame: NSZeroRect];
+
+  if (self != nil)
+    {
+      FOR_IN(NSMutableArray*, array, rows)
+        {
+          [_rows addObject: array];
+        }
+      END_FOR_IN(rows);
+    }
+  
+  return self;
+}
+
++ (instancetype) gridViewWithNumberOfColumns: (NSInteger)columnCount rows: (NSInteger)rowCount
+{
+  NSUInteger r = 0;
+  NSUInteger c = 0;
+  NSMutableArray *rows = [[NSMutableArray alloc] initWithCapacity: rowCount]; 
+
+  for (r = 0; r < rowCount; r++)
+    {
+      NSMutableArray *col = [NSMutableArray arrayWithCapacity: columnCount];
+      for (c = 0; c < columnCount; c++)
+        {
+          NSGridCell *gc = [[NSGridCell alloc] init];
+          [col addObject: gc];
+          RELEASE(gc);
+        }
+      [rows addObject: col];
+    }
+  
+  return AUTORELEASE([self gridViewWithViews: rows]);
+}
+
++ (instancetype) gridViewWithViews: (NSArray *)rows
+{
+  return [[self alloc] initWithViews: rows];
+}
+
+- (NSInteger) numberOfRows
+{
+  return [_rows count];
+}
+
+- (NSInteger) numberOfColumns
+{
+  return [_columns count];
```

Comment:
The only place where you set _columns is in the initWithCoder: method.

Reply:
True.  Implementation is ongoing.  I still need to implement the methods which are used to programmatically add rows and columns.

Reply:
done